### PR TITLE
Style client and add functionality

### PR DESCRIPTION
**Pull request by avincenthill**

### CODE REVIEW

File: client/src/components/Header.jsx [hunk 1]
```diff
@@ -2,13 +2,24 @@ import React from 'react';
...
     super(props);
   }
 
-  renderRatingHistogram(reviews) {
+  componentDidMount() {
+    // check if inview and if so animate
+    setInterval(() => {
+      if ($('#scrollTarget').visible()) {
+        // wildcard jquery selector
+        $('[id^=histBar]').addClass(styles.animated);
```
Comment (mattminwoolee) [inline client/src/components/Header.jsx:17]: this looks cool. would love to hear about what this is later!

File: client/src/components/Mentions.jsx [hunk 2]
```diff
@@ -4,28 +4,40 @@ import styles from '../styles/Mentions.css';
...
+    const newExpanded = !this.state.expanded;
+    this.setState({
+      expanded: newExpanded,
+    });
   }
 
   render() {
-    const state = this.props.getState();
-    const { reviews } = state;
+    const mentionPhrases = ['lorem', 'ipsum', 'is', 'simply', 'dummy', 'text', 'of', 'the', 'printing', 'and', 'typesetting', 'industry', 'lorem', 'ipsum', 'is', 'simply', 'dummy', 'text', 'of', 'the', 'printing', 'and', 'typesetting', 'industry'];
```
Comment (mattminwoolee) [inline client/src/components/Mentions.jsx:20]: Is every product going to have these mentionPhrases?

File: client/src/components/Review.jsx [hunk 3]
```diff
@@ -6,32 +6,33 @@ class Review extends React.Component {
...
     if (review.verifiedPurchase) {
       isVerified = 'Verified Purchase';
     }
     const starArr = this.props.renderStarRating(review.stars, 1.5);
-
     return (
       <React.Fragment>
         <div className={styles.review}>
           <div className={styles.profileContainer}><img className={styles.profile} src='./img/profile.jpg'></img> {username}</div>
           <div><a className={styles.gold}><b>{starArr}</b> </a><b>{review.title}</b></div>
```
Comment (soniaroh) [inline client/src/components/Review.jsx:29]: fix indentations

**Review by soniaroh — APPROVED** [review]

File: client/src/components/TopReviews.jsx [hunk 4]
```diff
@@ -12,7 +12,7 @@ class TopReviews extends React.Component {
     const reviewElements = reviews.map((review, index) => {
       // TBD toggle show reviews all on button press
       if (index < 5) {
-        return <Review key={index} review={review} renderStarRating={this.props.renderStarRating} />;
+        return <Review incrementHelpfulness={this.props.incrementHelpfulness} key={index} review={review} renderStarRating={this.props.renderStarRating} />;
```
Comment (mattminwoolee) [inline client/src/components/TopReviews.jsx:15]: Nice

File: client/src/components/Mentions.jsx (hunk 2 above)
Comment (mattminwoolee) [reply client/src/components/Mentions.jsx:20]: Is every product going to have these mentionPhrases?

File: client/src/styles/Header.css [hunk 5]
```diff
@@ -120,18 +115,49 @@
...
   timing-function: ease-in;
 }
 
 @keyframes stretchInLeft {
   from { width: 0%; }
+}
+
+:local .blueLarge:hover {
+  cursor: pointer;
+  color: #C15926 !important;
```
Comment (mattminwoolee) [inline client/src/styles/Header.css:137]: Would love to know what !important is used for.

File: server/index.js [hunk 6]
```diff
@@ -44,6 +44,15 @@ class Server {
         res.status(202).send(data);
       });
     });
+
+    // increment helpfullness
+    this.app.get('/helpful/*', bodyParser.json(), (req, res) => {
+      const reviewId = req.originalUrl.split('/')[2]; // get reviewId from from url
+      db.incrementHelpfulness(reviewId, (err, data) => {
+        if (err) return console.error(err);
+        res.status(202).send();
```
Comment (mattminwoolee) [inline server/index.js:53]: I think express sends a default status code for you on successful data retrieval so status(202) should not be needed.

File: client/src/components/Header.jsx (hunk 1 above)
Comment (avincenthill) [reply client/src/components/Header.jsx:17]: this is jquery to detect if target is in viewport and add class to animate histogram

File: client/src/components/Mentions.jsx [hunk 7]
```diff
@@ -4,28 +4,40 @@ import styles from '../styles/Mentions.css';
...
-          <div className={styles.mentionItem}>mentionmention</div>
-          <div className={styles.mentionItem}>mention</div>
-          <div className={styles.mentionItem}>mention</div>
-          <div className={styles.mentionItem}>mention</div>
-          <div className={styles.mentionItem}>mentionmention</div>
-          <div className={styles.mentionItem}>meon</div>
-          <div className={styles.mentionItem}>mentmention</div>
-          <div className={styles.mentionItem}>mention</div>
-          <div className={styles.mentionItem}>mtion</div>
+          {mentions}
```
Comment (mattminwoolee) [inline client/src/components/Mentions.jsx:37]: nice

**Review by mattminwoolee — APPROVED** [review]

File: client/src/components/Mentions.jsx (hunk 2 above)
Comment (avincenthill) [reply client/src/components/Mentions.jsx:20]: at the moment yes, this is another feature that is a reach goal

File: client/src/styles/Header.css (hunk 5 above)
Comment (avincenthill) [reply client/src/styles/Header.css:137]: this is used to force overwrite css or refuse to inherit css from below I think

File: server/index.js (hunk 6 above)
Comment (avincenthill) [reply server/index.js:53]: thanks will change

File: client/src/components/Mentions.jsx [hunk 8]
```diff
@@ -4,28 +4,40 @@ import styles from '../styles/Mentions.css';
...
+    });
   }
 
   render() {
-    const state = this.props.getState();
-    const { reviews } = state;
+    const mentionPhrases = ['lorem', 'ipsum', 'is', 'simply', 'dummy', 'text', 'of', 'the', 'printing', 'and', 'typesetting', 'industry', 'lorem', 'ipsum', 'is', 'simply', 'dummy', 'text', 'of', 'the', 'printing', 'and', 'typesetting', 'industry'];
+
+    let mentions = mentionPhrases.map((phrase, index) => {
+      return <div key={index} className={styles.mentionItem}>{phrase}</div>;
```
Comment (michchen58) [inline client/src/components/Mentions.jsx:23]: does ESLint complain that "Do not use array index in keys"? i keep getting that

**Review by michchen58 — APPROVED** [review]

File: client/src/components/Mentions.jsx (hunk 8 above)
Comment (avincenthill) [reply client/src/components/Mentions.jsx:23]: I don't see problems on this line, but I'm using the AirBnB eslint config, not HR

**Pull request merged**